src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/service/SupplierPricingService.kt

```diff
@@ -34,15 +43,40 @@ class SupplierPricingService(val locationRepository: LocationRepository, val pri
   fun addPriceForSupplier(supplier: Supplier, fromAgencyId: String, toAgencyId: String, price: Money) {
     val (fromLocation, toLocation) = getFromAndToLocationBy(fromAgencyId, toAgencyId)
 
-    priceRepository.save(Price(supplier = supplier, fromLocation = fromLocation, toLocation = toLocation, priceInPence = price.pence, effectiveYear = 2020))
+    priceRepository.save(
+      Price(
+        supplier = supplier,
+        fromLocation = fromLocation,
+        toLocation = toLocation,
+        priceInPence = price.pence,
+        effectiveYear = 2020
+      )
+    )
+
+    SecurityContextHolder.getContext().authentication?.let {
```

Comment:
Accessing the security context here the way we are is probably OK.  We could probably look at using Aspects for this but I think for now we can just stick to this.